src/GraphQL/Execution/SubscriptionExecutionStrategy.cs

```diff
@@ -122,33 +122,42 @@ public override async Task<ExecutionResult> ExecuteAsync(ExecutionContext contex
             return null;
         }
 
+        // preserve required information from the context
+        var preservedContext = CloneExecutionContext(context, default);
+
         // cannot throw an exception here
         return source
             .SelectCatchAsync(
                 async (value, token) =>
                 {
                     // duplicate context to prevent multiple event streams from sharing the same context,
                     // and clear errors/metrics/extensions. Free array pool leased arrays after execution.
-                    using var childContext = CloneExecutionContext(context, token);
+                    using var childContext = CloneExecutionContext(preservedContext, token);
```

Comment:
🤔 Double clone here and for error. Why did you add `preservedContext` ?

Reply:
mostly so that the disposed service provider during the initial execution can be cleared, so no references are being held to it unnecessarily.  besides that, references to the other unused fields (output extensions, metrics, etc) are also released to the GC.

It is still necessary to clone for each processed data event so that it does not 'corrupt' original clone.

Reply:
I understand general intention about clearing service provider and other stuff. I don't understand about **double** cloning.

Reply:
Because the original `context` is held indefinitely until the subscription ends.  It is not desirable to hold a reference to the service provider (which may be scoped) or other transient information.  Nor can we clear it at this point because the DocumentExecuter has not finished.  So we clone it prior to SelectCatchAsync so that only the references we need to hold are held.

With the changes you can hold your mouse over the delegate and see that it says "captured references" are "this" and "preservedContext" now, and not "context".

So far so good.  Then we move on to execution of processed data events.  Events are processed asynchronously and, in the case that processing does not occur synchronously (at some point it actually awaits something), it can occur simultaneously.  So it is necessary for each processing event to own its own context, and being able to create errors, output extensions and/or a service scope for each execution.  So it is cloned prior to the execution.

Note that it is not cloned twice per data event.  It is cloned once initially, and then cloned from the clone for each subsequent execution.

If we do not add the "additional" clone, then we run into the issue you pinned previously, where the SelectCatchAsync is going to hold a reference to a disposed service provider including all of the disposed objects that were created during initial execution.

Reply:
>Note that it is not cloned twice per data event. It is cloned once initially, and then cloned from the clone for each subsequent execution.

🤦‍♂️  Understood.